### ISSUE TITLE
Figure directives

### ISSUE THREAD

**Issue by thedrow** [sphinx-contrib/images#15]
Can we add support for figure directives as well as image directives?

**Comment by jonascj** [comment]
Could you elaborate a little? 

There is the [numfig](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-numfig) configuration option which adds numbers to all "figures", tables etc. if they have a caption.

There is also the [figure](https://docutils.sourceforge.io/docs/ref/rst/directives.html#figure) directive. 

Do you just wish the lightbox2 css applied to images inserted/loaded through the `figure: whatever.jpg` directive?

**Comment by thedrow** [comment]
> Do you just wish the lightbox2 css applied to images inserted/loaded through the `figure: whatever.jpg` directive?

Yes.

**Comment by jonascj** [comment]
Great idea, I could easily have wished for that myself in the past. Feel free to implement it and make a PR, otherwise I'll have a look at it a few weeks from now.

**Comment by jonascj** [comment]
How time flies when you have a baby and a 3-year-old! Please accept my apologies for not attending to the PRs and issues in a timely fashion. I'll attend to them at the latest on August 29th and 30th (I have another project with deadline on August 28th).

**Comment by icq8680** [comment]
@jonascj I think the main concern of adding lightbox to figures is that lightbox label does not have a translatable text, i.e. if you add a caption to your LB image, you won't be able to translate the phrase

**Comment by SilverRainZ** [comment]
Any news?

**Comment by jonascj** [comment]
@SilverRainZ No, I haven't looked into it yet. You are welcome to submit a PR if you can find a way to implement it. See also #20 :-)

**Comment by jonascj** [comment]
Some findings and/or realizations which might be worth remembering, going forward with this:

 - Typical HTML markup for a figure (after building):
  ```
  <figure class="align-default" id="id1">                                         
      <span id="some-fig-label"></span>                                               
      <img  alt="_images/python.png" src="_images/python.png" />
      <figcaption>                                                                    
      <p>
          <span class="caption-number">Fig. 1 </span>
          <span class="caption-text">The Python logo.</span>
          <a class="headerlink" href="#id    1" title="Permalink to this image">¶</a>
      </p>          
      </figcaption>                                                                   
  </figure>
  ```
   All that is needed to turn this into a lightbox2 image is to wrap the `<img>` 
   tag in `<a data-lightbox="[unique id]" data-title="The Python logo." href="_images/python.png">`.
 
- At present neither `.. image::` or `.. thumbnail::` directives can have a label and be referenced. The reason being neither have a title or caption associated with them. So if referenced what should the reference show? `Fig. <num>` is not meaningful since it has no number. `The Python logo` is not meaningful either since it has no title.
 
- There seem to be two somewhat equivalent objectives / solutions: 1) Make the `.. thumbnail::` directive numbered so it can be referenced, 2) Wrap the `.. figure::` directive's `<img>` tag in a lightbox link (see above).

**Comment by thedrow** [comment]
I don't see why thumbnails shouldn't have a unique id, even if it is a random one.

**Comment by jonascj** [comment]
#27 Indicates there is currently no way of referencing `.. thumbnail::` directives, explicit targets do not work (see above or #27). That is a bug imo and we should at least fix that.

At the same time I think we should make the `.. thumbnail::` directive behave like `.. figure::`, if `numfig = True` thumbnails should be numbered just as figures (maybe with a sphinxcontrib.images configuration option to turn on/off this feature). I suppose it could be done two ways: 1) Utilize the figure directive when producing thumbnails, 2) create/register a counter with the `.. thumbnail::` directive.
